management_app/views/course.py

```diff
@@ -59,57 +60,44 @@ def offerings():
-    #         combine_with = None
-    #     else:            
-    #         combine_with = str(row[4]).strip().replace(' ', '')
-        
-
-    #     db.execute(
-    #         'INSERT INTO courses (course_title_id, course_title, units, course_level, combine_with)'
-    #         ' VALUES (?, ?, ?, ?, ?)',
-    #         (course_title_id, course_title, units, course_level, combine_with)
-    #     )
-    #     db.commit()
-
-
-    # TODO: show "combine_with" column in the front end so that admin will remember to edit it if he needs to
-    # TODO: check: the "unit" column of the courses file only has single values (in the unit column, I won’t get values like 2-4 or 4+2), if not: generate warning and reject add/edit
-
-    db = get_db()
+    db = get_db()    
     courses = db.execute('SELECT * FROM courses').fetchall()
-
     return render_template('courses/catalog.html', courses=courses)
 
 
 @courses.route('/data-templates/<filename>', methods=['GET'])
 @login_required
 def download_template(filename):
-
-    # TODO: do template Prefill: 
-    # Schedule_teaching template prepopulate: year, quarter, usename, ucinetid, 
-        # course_type： Lec, course_sec: A (each prodessor 3 rows: fall, winter, spring)
-    # courses template: no need to prepopulate
+    db = get_db()
+    res = db.execute('SELECT COUNT(*) FROM scheduled_teaching').fetchone()
+    # Prepopulate schedule_teaching template after the first time upload
+    # Should include: year, quarter, ucinetid, course_sec
+    # Each prodessor has 3 rows: fall, winter, spring
+    if res != 0:
+        # get the latest academic year from scheduled_teaching
+        year = db.execute(
+            'SELECT DISTINCT year FROM scheduled_teaching ORDER BY year DESC LIMIT 1'
+        ).fetchone()['year']
+
+        ucinetids = get_ucinetid()
+        rows_data = []
+        for ucinetid in ucinetids:
+            rows_data.append([year, "Fall", ucinetid, "", "A", "", ""])
+            rows_data.append([year+1, "Winter", ucinetid, "", "A", "", ""])
+            rows_data.append([year+1, "Spring", ucinetid, "", "A", "", ""])
+
+        df1 = pd.DataFrame([["Scheduled_teaching", "quarter", "1 (Fall), 2 (Winter), 3 (Spring), 4 (Summer)"], ["", "course_title_id", "CS143B"], ["", "course_sec", "A"], ["", "num_of_enrollment", "180"], ["", "offload_or_recall_flag", "1 for \"offload / recall\" or 0"]], 
+            columns=[" ", "column name", "column values"])
+
+        df2 = pd.DataFrame(rows_data,
+            columns=["year", "quarter", "user_UCINetID",	"course_title_id", "course_sec", "num_of_enrollment", "offload_or_recall_flag"])    
+
+        with pd.ExcelWriter(os.path.join(BASE_DIR, DOWNLOAD_FOLDER, 'scheduled_teaching.xlsx'), mode='w') as writer:  
+            df1.to_excel(writer, sheet_name='Examples for column value', index=False)
+            df2.to_excel(writer, sheet_name='Scheduled_teaching', index=False)
 
     if request.method == 'GET':
         return download_file(filename)
```

Comment:
When there is no data in the system, download is failed.

Reply:
Fixed.